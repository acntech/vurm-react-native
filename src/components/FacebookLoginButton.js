import { FacebookSocialButton } from "react-native-social-buttons";
import * as Facebook from "expo-auth-session/providers/facebook";
import { FacebookAuthProvider, signInWithCredential } from "firebase/auth";
import { useEffect, useState } from "react";
import { auth } from "../firebase/auth";
import { ResponseType } from "expo-auth-session";
const FacebookLoginButton = ({
  buttonViewStyle,
  setUser,
  setDisabled,
  disabled,
}) => {
  const [request, response, promptAsync] = Facebook.useAuthRequest({
    responseType: ResponseType.Token,
    clientId: "498173242097723",
  });
  const handlePress = () => {
    setDisabled(true);
    promptAsync().finally(() => {
      setDisabled(false);
    });
  };

  useEffect(() => {
    if (response?.type === "success") {
      // Build Firebase credential with the Facebook access token.
      const { access_token } = response.params;

      const credential = FacebookAuthProvider.credential(access_token);
      signInWithCredential(auth, credential)
        .then(() => {
          setUser(auth.currentUser);
        })
        .catch((error) => {
          console.log(error);
        });
    }
  });
  return (
    <FacebookSocialButton
      buttonViewStyle={buttonViewStyle}
      onPress={handlePress}
      disabled={disabled}
    />
  );
};

export default FacebookLoginButton;
