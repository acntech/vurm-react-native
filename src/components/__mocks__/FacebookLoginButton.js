import { FacebookSocialButton } from "react-native-social-buttons";
import { auth } from "../../firebase/auth";
const FacebookLoginButton = ({
  buttonViewStyle,
  setUser,
  setDisabled,
  disabled,
}) => {
  const handlePress = () => {
    setUser({ uid: "testUid", displayName: "Test Person" });
  };
  return (
    <FacebookSocialButton
      buttonViewStyle={buttonViewStyle}
      onPress={handlePress}
      disabled={disabled}
    />
  );
};

export default FacebookLoginButton;
