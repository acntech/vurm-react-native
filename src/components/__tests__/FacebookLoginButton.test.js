import React from "react";
import { render } from "@testing-library/react-native";
import FacebookLoginButton from "../FacebookLoginButton";

jest.mock("expo-auth-session/providers/facebook", () => {
  const originalModule = jest.requireActual(
    "expo-auth-session/providers/facebook"
  );

  //Mock the default export and named export 'foo'
  return {
    __esModule: true,
    ...originalModule,
    useAuthRequest: jest.fn(() => [undefined, undefined, undefined]),
  };
});

describe("<FacebookLoginButton />", () => {
  it("renders correctly", () => {
    // Arrange & Act
    const tree = render(<FacebookLoginButton />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
