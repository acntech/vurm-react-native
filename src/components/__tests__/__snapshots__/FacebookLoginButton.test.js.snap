// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacebookLoginButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#485a96",
      "borderColor": "#fff",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "margin": 5,
      "marginBottom": 0,
      "opacity": 1,
      "position": "relative",
      "width": 220,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 25,
        "marginLeft": 15,
        "padding": 10,
        "resizeMode": "stretch",
        "width": 25,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  >
    Sign in with Facebook
  </Text>
</View>
`;
